### PR TITLE
LPD-26897 - Keep selected data set checked when opening FDS Admin Item Selector (findings)

### PR DESCRIPTION
**Pull request by juanjofgliferay**
## Bug
https://liferay.atlassian.net/browse/LPD-26897

> [!WARNING]  
> Spaghetti code inside

## Issue
It is possible to select a Data Set when dropping the fragment in a page. However, it is not possible to open the Data Set selection modal and have the one we are using marked as checked in the list.

## Findings
There are lot of things that are not working as expected or, at least, I didn't find a way to make them work as expected.
I'll try to summarize all the things missing.

### FDSAdminItemSelector.tsx
- I didn't find a standardized way to get any data from the `Modal.js` component.
- Even if there is a way to pass the selected item, we are not providing the information to the FDS component. We need to add a `selectedItems` prop to the component and fill it properly. 
- I've tested passing properties through the `FDSAdminItemSelectorDisplayContext`, like the `itemSelectedEventName` and adding a custom eventListener, but it didn't work
- I've tried retrieving/passing also the `classPK` property, but it always result in getting a `0` as value.

### itemSelector.js / openItemSelector.js
The itemSelector component knows which is the **selectedItem** but it is not passing the information when it triggers the open modal callback.

### Modal.js
The `Modal.js` component from `frontend-js-web` accepts a `selectedData` property but it is opinionated and there is nothing we can do with it (easily). 
It seems that there is no other way to pass data from the modal to the `<Component >` that we are going to render inside it.

## PoC result
After tinkering a bit with all involved components a make my (dirty) way to pass data from the `itemSelector` to the `FDSAdminItemSelector` component trough the `Modal` component by adding the `selectedItem` as part as the `iframe` properties and parsing them when the modal is openened.

## Expected result
https://github.com/user-attachments/assets/dd7e6beb-fb4c-4aeb-a9a9-fc3d513934db

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/display/context/FDSAdminItemSelectorDisplayContext.java [hunk 1]
```diff
@@ -41,6 +48,18 @@ public long getClassNameId() {
 		return PortalUtil.getClassNameId(getClassName());
 	}
 
+	public String getItemSelectedEventName() {
+		return _itemSelectedEventName;
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/display/context/FDSAdminItemSelectorDisplayContext.java:52]: Tried `itemSelectedEventName`, also provided in [itemSelector](https://github.com/liferay-frontend/liferay-portal/pull/4362/files#diff-b094c5f25620284eeb36a69b9802d581e15de4acf2397664ca9b1ff1dd398542R69) to use as eventListener.

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/display/context/FDSAdminItemSelectorDisplayContext.java [hunk 2]
```diff
@@ -19,8 +21,13 @@
 public class FDSAdminItemSelectorDisplayContext {
 
 	public FDSAdminItemSelectorDisplayContext(
-		HttpServletRequest httpServletRequest) {
+		HttpServletRequest httpServletRequest,
+		String itemSelectedEventName,
+		InfoItemItemSelectorCriterion infoItemItemSelectorCriterion) {
 
+		_classPK = ParamUtil.getLong(httpServletRequest, "refererClassPK");
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/display/context/FDSAdminItemSelectorDisplayContext.java:28]: The `itemSelector` sends a `refererClassPK` param through the [itemSelectorURL param](https://github.com/liferay-frontend/liferay-portal/pull/4362/files#diff-b094c5f25620284eeb36a69b9802d581e15de4acf2397664ca9b1ff1dd398542R70), but it is always set to `0`.

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/item/selector/FDSAdminItemSelector.tsx [hunk 3]
```diff
@@ -30,16 +30,34 @@ const views = [
...
-	const [selectedItem, setSelectedItem] = useState<ISelectedItem>();
+	className: string;
+	classNameId: string;
+	classPK: string;
+	namespace: string;
+}
+) => {
+	const selectedId = window.frameElement && window.frameElement?.getAttribute('classPK');
+
+	const preselectedItem = {
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/item/selector/FDSAdminItemSelector.tsx:48]: Build the `preselectedItem` from the props attached to the `<iframe>` in the [Modal](https://github.com/liferay-frontend/liferay-portal/pull/4362/files#diff-15984289d46b19a7d68a56b2d41c72a99fa2307cb9628c43f2d0f32f2b47129fR262)

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/components/ItemSelector.js [hunk 4]
```diff
@@ -56,13 +56,21 @@ export default function ItemSelector({
...
+			transformValueCallback,
+			selectedItem,
+		});
 		openItemSelector({
 			callback: onItemSelect,
 			eventName: eventName || `${config.portletNamespace}selectInfoItem`,
 			itemSelectorURL: itemSelectorURL || config.infoItemSelectorURL,
 			modalProps,
 			transformValueCallback,
+			selectedData: selectedItem,
```
Comment (juanjofgliferay) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/components/ItemSelector.js:73]: As `selectedItem` is available from this component, pass it to the modal.

**Comment by markocikos** [comment]
Followup in https://github.com/liferay-frontend/liferay-portal/pull/4382

**Pull request closed without merging**